src/libraries/Common/src/Interop/Linux/procfs/Interop.ProcFsStat.TryReadStatusFile.cs

```diff
@@ -149,21 +150,52 @@ internal static bool TryParseStatusFile(string statusFilePath, out ParsedStatus
             return true;
         }
 
-        private static bool TryReadFile(string filePath, ReusableTextReader reusableReader, [NotNullWhen(true)] out string? fileContents)
+        private static bool TryReadFile(string path, [NotNullWhen(true)] out string? contents)
         {
+            Debug.Assert(!string.IsNullOrEmpty(path));
+
+            byte[]? bytes = null;
+            const int size = 4096;
+
             try
             {
-                using (var fileStream = new FileStream(filePath, FileMode.Open, FileAccess.Read, FileShare.Read, bufferSize: 1, useAsync: false))
+                using var fileStream = new FileStream(path, FileMode.Open, FileAccess.Read, FileShare.Read, bufferSize: 1, useAsync: false);
+
+                bytes = ArrayPool<byte>.Shared.Rent(size);
+                int count = 0;
+
+                while (true)
                 {
-                    fileContents = reusableReader.ReadAllText(fileStream);
-                    return true;
+                    int read = fileStream.Read(bytes, count, bytes.Length - count);
+                    if (read == 0)
+                    {
+                        contents = Encoding.UTF8.GetString(bytes, 0, count);
+                        return true;
+                    }
+
+                    count += read;
+                    if (count >= bytes.Length)
+                    {
+                        byte[] temp = ArrayPool<byte>.Shared.Rent(bytes.Length * 2);
+                        Array.Copy(bytes, 0, temp, 0, bytes.Length);
```

Comment:
```suggestion
                        Array.Copy(bytes, temp, bytes.Length);
```
The simpler overload is faster these days (it was not always like that).

Reply:
Thanks. Fixed with other feedback.